Integrations/src/main/java/io/deephaven/integrations/learn/FutureOffset.java

```diff
@@ -0,0 +1,43 @@
+package io.deephaven.integrations.learn;
+
+import io.deephaven.base.verify.Require;
+import org.jpy.PyObject;
+
+/**
+ * FutureOffset pairs a deferred calculation with an offset, an index that indicates the location in the calculated result.
+ */
+public class FutureOffset {
+
+    private final Future future;
+    private final int offset;
+
+    /**
+     * Creates a new FutureOffset.
+     *
+     * @param future    deferred calculation.
+     * @param offset    location in the calculated result.
+     */
+    public FutureOffset(Future future, int offset) {
```

Comment:
Does this need to be public?

Reply:
Yes, some of it's methods are used from the query string in Python.